src/compile/data/nonpositivenullfilter.ts

```diff
@@ -23,32 +23,24 @@ export namespace nonPositiveFilter {
     }, {} as Dict<boolean>);
   }
 
-  export function parseFacet(model: FacetModel) {
-    const childDataComponent = model.child().component.data;
+  export function mergeIfCompatible(dataComponent: DataComponent, childDataComponents: DataComponent[]) {
+    const nonPosComponent = childDataComponents.reduce((collector, childData) => {
+      extend(collector, childData.nonPositiveFilter);
+      return collector;
+    }, duplicate(dataComponent.nonPositiveFilter));
```

Comment:
when would parent have `nonPositiveFilter`?  This only comes from log scale and none of facet / concat / repeat / layer can have log scale.